service/auth.ts

```diff
@@ -13,6 +13,30 @@ import {
 } from "proto/auth_pb";
 import client from "service/client";
 
+type CreateUserReq = {
+  email: string,
+  username: string,
+  password: string
+}
+
+type CreateUserRes = {
+  email: string,
+  username: string,
+  id: number
+}
+
+export type HttpError = {
+  status_code: number,
+  errors: [string],
```

Comment:
Probably a nitpick: but we tend to prefer interfaces where possible especially for typing shapes of objects since they tend to have slightly better type checking performance when you need to start combining them!